### PR TITLE
Readme content updates, prepping for OCI adoption

### PR DESCRIPTION
**Pull request by SteveLasker**
Readme content updates, prepping for OCI adoption

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -138,13 +139,13 @@ The following sample defines a new Artifact Type of **Acme Rocket**, using `appl
 
   ```sh
   oras push localhost:5000/hello-artifact:v2 \
-  --manifest-config /dev/null:application/vnd.acme.rocket.config.v1+json \
+  --manifest-config /dev/null:application/vnd.acme.rocket.config \
     artifact.txt:application/vnd.acme.rocket.layer.v1+txt
```
[the remark this answers is not in the document]
Comment (SteveLasker) [reply README.md]: @jdolitsky see updates, including text/plain, and a better example for generating a tar from a directory.

File: README.md [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
 ![ORAS](./oras.png)
 
-[Registries are evolving as Cloud Native Artifact Stores](https://stevelasker.blog/2019/01/25/cloud-native-artifact-stores-evolve-from-container-registries/). To enable this goal, Microsoft has donated ORAS as a means to enable various client libraries with a way to push artifacts to [OCI Spec Compliant](https://github.com/opencontainers/image-spec) registries.
+[Registries are evolving as Cloud Native Artifact Stores](https://stevelasker.blog/2019/01/25/cloud-native-artifact-stores-evolve-from-container-registries/). To enable this goal, Microsoft has donated ORAS as a means to enable various client libraries with a way to push [OCI Artifacts][artifacts] to [OCI Spec Compliant](https://github.com/opencontainers/oci-conformance) registries.
```
Comment (sajayantony) [inline README.md:9]: OCI Spec `conformant` registries?

File: README.md [hunk 3]
```diff
@@ -109,8 +109,11 @@ See [Supported Registries](./implementors.md) for registry specific authenticati
 ### Pushing Artifacts with Single Files
 
 Pushing single files involves referencing the unique artifact type and at least one file.
+Defining an Artifact uses the `config.mediaType` as the unique identifier. If a config object is provided, the `mediaType` extension defines the config filetype. If a null config is passed, the config extension is removed.
```
Comment (sajayantony) [inline README.md:112]: consider rewording unique identifier to `the artifact type` identifier.

File: README.md [hunk 4]
```diff
@@ -157,20 +160,28 @@ The [OCI distribution-spec][distribution-spec] provides for storing optional con
...
+    artifact.txt:text/plain
   ```
 
 ### Pushing Artifacts with Multiple Files
 
-Just as container images support multiple "layers", represented as blobs, ORAS supports pushing multiple files. The file type is up to the artifact author. You can push `.tar` to represent a collection of files or individual files like `.yaml`, `.txt` or whatever your artifact should be represented as. Each blob (layer) type should have a `mediaType` to represent the type of blob. See [OCI Artifacts][artifacts] for more details.
+Just as container images support multiple "layers" represented as blobs, ORAS supports pushing multiple layers. The layer type is up to the artifact author. You may push `.tar` representing a collection of files, individual files like `.yaml`, `.txt` or whatever your artifact should be represented as. Each layer type should have a `mediaType` representing the type of blob content.
+In this example, we'll push a collection of files.
+
+- A single root file (`artifact.txt`) that represents overview content that might be displayed as a repository overview
```
Comment (sajayantony) [inline README.md:171]: Recommend eliding the word root to avoid confusion with rootfs.

File: README.md (hunk 2 above)
Comment (SteveLasker) [reply README.md:9]: Was trying to align with the "conformance tests", but this reads better

**Comment by sajayantony** [comment]
> Was trying to align with the "conformance tests", but this reads better 

My only concern with compliance is that it almost sounds legal. Basically the definition of conformance/compliance is the point.

**Pull request merged**